### ISSUE TITLE
Remove `android`, `dinput`, `hid`, `mfi`, `qnx`, `x`, `xinput` from the autoconfig directory

### ISSUE THREAD

**Issue by davidhedlund** [flathub/org.libretro.RetroArch#304]
The AppImage supports the following controller drivers via the **Settings -> Drivers -> Controller** menu:

- `linuxraw`
- `parport`
- `sdl2`
- `udev`

The directory `/app/share/libretro/autoconfig` contains sub-directories for various controller configurations, including:

- `android`
- `dinput`
- `hid`
- `linuxraw`
- `mfi`
- `parport`
- `qnx`
- `sdl2`
- `udev`
- `x`
- `xinput`

The sub-directories not supported by the AppImage settings (`android`, `dinput`, `hid`, `mfi`, `qnx`, `x`, `xinput`) could potentially be removed if they are not needed for any specific use cases or configurations. However, these directories may be included to support a broader range of devices and configurations beyond the default settings. Removing them would depend on whether you want to limit the configurations to only those explicitly supported by the AppImage settings or maintain broader compatibility.

**Comment by davidhedlund** [comment]
* https://github.com/libretro/docs/pull/982

**Comment by RobLoach** [comment]
They don't take up much file space. Is the goal to make the user interface menu smaller?

**Comment by davidhedlund** [comment]
> They don't take up much file space.

That is true, but they may distract people making them believe that these drivers are supported.

Do you agree with my statement in the top post? That:
> Also, in free systems like `flatpak remote-modify --subset=floss flathub` which retroarch is downloaded from, it's unusual to encounter data dedicated to proprietary drivers like `xinput` and `dinput`.

**Comment by RobLoach** [comment]
I agree with your statement above, yes. But it still doesn't hurt to have them. We could add a removal script, but 🤷

While those systems may be proprietary, the config files for them are not. They're MIT:
https://github.com/libretro/retroarch-joypad-autoconfig/blob/master/COPYING
